src/Security/InheritedPermissions.php

```diff
@@ -610,4 +699,34 @@ protected function getViewerGroupsTable()
         $table = DataObject::getSchema()->tableName($this->baseClass);
         return "{$table}_ViewerGroups";
     }
-}
+
+    /**
+     * Gets the permission from cache
+     *
+     * @param $cacheKey
+     * @return mixed
+     */
+    protected function getCachePermissions($cacheKey)
+    {
+        if (isset($this->cachePermissions[$cacheKey])) {
+            return $this->cachePermissions[$cacheKey];
+        }
+
+        if ($this->cacheService) {
+            return $this->cacheService->get($cacheKey);
```

Comment:
I think we should have the persistent cache warm the non-persistent cache here. I'll update. :)